pkg/k8shandler/visualization.go

```diff
@@ -283,17 +280,40 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateKibanaDeployment(prox
 			current, different := isDeploymentDifferent(current, kibanaDeployment)
 
 			// Check trustedCA certs have been updated or not by comparing the hash values in annotation.
-			newTrustedCAHashedValue, err := calcTrustedCAHashValue(kibanaTrustBundle)
+			newTrustedCAHashValue, err := calcTrustedCAHashValue(kibanaTrustBundle)
 			if err != nil {
 				return fmt.Errorf("unable to calculate trusted CA value. E: %s", err.Error())
 			}
-			trustedCAHashedValue, _ := current.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName]
-			if trustedCAHashedValue != newTrustedCAHashedValue {
+			trustedCAHashValue, _ := current.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName]
+			if trustedCAHashValue != newTrustedCAHashValue {
 				different = true
 				if kibanaDeployment.Spec.Template.ObjectMeta.Annotations == nil {
 					kibanaDeployment.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
 				}
-				kibanaDeployment.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName] = newTrustedCAHashedValue
+				kibanaDeployment.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName] = newTrustedCAHashValue
+			}
+
+			// Check secret hash has been updated or not
+			for _, secretName := range []string{"kibana", "kibana-proxy"} {
```

Comment:
this is to fix the logic that was being used before where we deleted the kibana pods in the case of the `kibana` or `kibana-proxy` secrets being updated
ref: https://github.com/openshift/cluster-logging-operator/pull/351

Reply:
ok, so fluentd was already handling this properly?

Reply:
fluentd doesn't need to be restarted in the case of a cert (secret) redeploy because it will reload certificates on its own. only ES and Kibana need to be externally restarted.